be/src/exec/schema_scanner/schema_partitions_meta_scanner.cpp

```diff
@@ -79,12 +79,25 @@ Status SchemaPartitionsMetaScanner::start(RuntimeState* state) {
             auth_info.__set_user_ip(*(_param->user_ip));
         }
     }
-    TGetPartitionsMetaRequest partitions_meta_req;
-    partitions_meta_req.__set_auth_info(auth_info);
 
     // init schema scanner state
     RETURN_IF_ERROR(SchemaScanner::init_schema_scanner_state(state));
-    RETURN_IF_ERROR(SchemaHelper::get_partitions_meta(_ss_state, partitions_meta_req, &_partitions_meta_response));
+    int64_t table_id_offset = 0;
+    while (true) {
+        TGetPartitionsMetaRequest partitions_meta_req;
+        partitions_meta_req.__set_auth_info(auth_info);
+        partitions_meta_req.__set_start_table_id_offset(table_id_offset);
+        TGetPartitionsMetaResponse partitions_meta_response;
+        RETURN_IF_ERROR(SchemaHelper::get_partitions_meta(_ss_state, partitions_meta_req,
+                                                          &partitions_meta_response));
+        _partitions_meta_vec.insert(_partitions_meta_vec.end(),
+                                    partitions_meta_response.partitions_meta_infos.begin(),
+                                    partitions_meta_response.partitions_meta_infos.end());
+        table_id_offset = partitions_meta_response.next_table_id_offset;
+        if (!table_id_offset) {
```

Comment:
check `partitions_meta_response.__isset.next_table_id_offset`?

Reply:
if not set, it will be zero

Reply:
explicit check isset instead of using the default value.